source/common/filter/tcp_proxy.h

```diff
@@ -30,7 +30,11 @@ namespace Filter {
   COUNTER(downstream_cx_tx_bytes_total)                                                            \
   GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
   COUNTER(downstream_cx_total)                                                                     \
-  COUNTER(downstream_cx_no_route)
+  COUNTER(downstream_cx_no_route)                                                                  \
+  COUNTER(upstream_pause_reading)                                                                  \
```

Comment:
If you take a look at generally the way we do stats, where possible we use cluster for the "upstream" stats. "Downstream" stats are taken care of by the filter itself. In this case its TCP proxy, but in future PRs you will obviously be doing the same type of thing in the HTTP connection manager. This is why today you see only downstream in the above stats and all the upstream stats are in the cluster stats. If you look at existing tcp proxy filter code you will see lots of uses of cluster stats for this purpose. I would just replicate that.

The main reason to do this is for consistency across protocols where possible.